client/pages/home.jsx

```diff
@@ -43,41 +45,159 @@ const data = [
 
 export default class Home extends React.Component {
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      error: '',
+      weight: '',
+      editing: false,
+      dayId: '',
+      loading: true
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+    this.loadData = this.loadData.bind(this);
+  }
+
   componentDidMount() {
-    // Fetch user data here
+    this.loadData();
+    this.setState({
+      loading: false
+    });
+  }
+
+  loadData() {
+    const date = new Intl.DateTimeFormat('en-US').format(new Date());
+    let dayData;
+    fetch(`/api/${token.userId}/weight`)
+      .then(res => res.json())
+      .then(data => {
+        for (const i in data) {
+          if (date === new Intl.DateTimeFormat('en-US').format(new Date(data[i].date))) {
+            dayData = data[i];
+          }
+        }
+        if (dayData) {
+          this.setState({
+            weight: dayData.weight,
+            dayId: dayData.historyId
+          });
+        } else {
+          this.setState({
+            error: 'Loading error'
+          });
```

Comment:
Here in these state updates is where you should set `loading` to `false`. You want to make sure you have a response from the server before updating loading in state